### DISCUSSION TITLE
How to send a message to particular client directly

### DISCUSSION THREAD

**Discussion by timepp** [moscajs/aedes#781] [Q&A] [answered]
Hi in our IOT case the server might send direct control message to client. Is that possible in Sedes? 

This is how IBM MQ support it: 
```

Do the task, Sending a message to an MQTT client using IBM MQ Explorer to send a message directly from IBM MQ to an MQTT client.

A message sent in this way to an MQTT client is called an unsolicited message. MQTT v3 clients receive unsolicited messages as publications with a topic name set. The telemetry (MQXR) service sets the topic name to the remote queue name.
```

One possible solution is that each client subscribe to a topic named by its own client ID, but I don't know if this is a good design.

**Comment by robertsLando** [comment, marked as answer]
You could try to do it as follow: 

```js
var client = broker.clients['clientId']

var packet = {
qos: 1,
retain: true,
topic: 'my/topic',
payload: 'Hello world'
}

client.publish(packet, cb)
```

**Comment by gnought** [comment]
@timepp can do, please checkout the `client.publish` method.
The `deliverQoS` is a internal method and may be changed, not to use it.

**Comment by robertsLando** [comment]
@gnought You right, I fixed my answer

**Comment by timepp** [comment]
Thanks!

**Comment by cybercoder** [comment]
> You could try to do it as follow:
> 
> ```js
> var client = broker.clients['clientId']
> 
> var packet = {
> qos: 1,
> retain: true,
> topic: 'my/topic',
> payload: 'Hello world'
> }
> 
> client.publish(packet, cb)
> ```

There is no ```clients``` property.

**Comment by robertsLando** [comment]
@cybercoder It's there but not typed I think
